experimental/auth-logic/endorsement-release/top_level.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/project-oak/transparent-release/experimental/auth-logic/wrappers"
+)
+
+const relationDeclarations = ".decl BuildPolicyAllowRelease(binary : Principal, hash : Sha256Hash)\n" +
+	".decl RealTimeNsecIs(time : Number)\n" +
+	".decl attribute hasPublicKey(hash : Sha256Hash)\n"
+
+// generateEndorsenent takes an application name, the path to an endorsement
+// file for that application, the path to a provenance file for that
+// application, and emits authorization logic code (as a string) that runs the
+// transparent release verification process.
+func generateEndorsement(authLogicInputs []string, appName, provenanceFilePath string) (string, error) {
+
+	var authLogicFileContents = ""
+	for _, authLogicInputFile := range authLogicInputs {
```

Comment:
I don't quite understand this part. Is it the input policy?

Reply:
This function combines the input policy with the outputs from the wrappers. The function can take any number of authorization logic policy files as inputs, so the for loop that this comment is about just concatenates all the authorization logic input files before concatenating them with the outputs from the provenance wrappers.

It's potentially useful to include multiple policy files as inputs because then some policy files can be shared across applications. In the example that generates the output that's checked in, there are two inputs:
 - `github_actions_policy.auth_logic`
 - `oak_endorsement_policy.auth_logic`

(With a slight change to the authorization logic compiler, we can just pass the outputs from the wrappers and these input policies directly to the authorization logic compiler. The only reason we can't already do this is that we need a tweak avoid collisions with relation declarations. I haven't fixed this yet because the authoriation logic compiler is being reimplemented in C++).

Reply:
Sounds good.